client/src/auth.tsx

```diff
@@ -99,6 +116,11 @@ export class Auth {
       return false;
     }
 
+    // schedule the tokens for renewal if they've not already done so
+    if (!Auth.renewTokenTimer) {
+      Auth.renewTokenTimer = setTimeout(Auth.renewTokens, 900000); // 15 minutes
```

Comment:
Is this supposed to be `setInterval` or is it just a one-time refresh?

Reply:
Also, if it is supposed to an interval, make sure to add a `clearInterval` in `logout`.

Reply:
the intention was that a refresh would be scheduled every time an auth check occurs (via `isAuthenticated`), but that's probably confusing and an interval makes more sense, so have updated.